packages/xstate-vue/src/useFsm.ts

```diff
@@ -0,0 +1,26 @@
+import { ref, onBeforeMount, onBeforeUnmount, Ref } from '@vue/composition-api';
+import { StateMachine, EventObject, interpret } from '@xstate/fsm';
+import { AnyEventObject } from 'xstate';
+
+export function useFsm<TC, TE extends EventObject = AnyEventObject>(
+  stateMachine: StateMachine.Machine<TC, TE, any>
+): {
+  state: Ref<StateMachine.State<TC, TE, any>>;
+  send: StateMachine.Service<TC, TE>['send'];
+  service: Ref<StateMachine.Service<TC, TE>>;
+} {
+  const state = ref<StateMachine.State<TC, TE, any>>(stateMachine.initialState);
+  const service = ref<StateMachine.Service<TC, TE>>(interpret(stateMachine));
+  const send = (event: TE | TE['type']) => service.value.send(event);
+
+  onBeforeMount(() => {
```

Comment:
is unmount guaranteed to run if this gets run?

Reply:
They correspond to the [Vue Lifecycle Hooks](https://vue-composition-api-rfc.netlify.com/api.html#lifecycle-hooks). ([See the diagram](https://vuejs.org/v2/guide/instance.html#Lifecycle-Diagram)) 

So it would live for the lifetime of the componennt, akin to the `useMachine` hook.